core/dataset.cpp

```diff
@@ -733,18 +735,19 @@ template <class Op, class A, class B>
 decltype(auto) apply_with_delay(const Op &op, A &&a, const B &b) {
   // For `b` referencing data in `a` we delay operation. The alternative would
   // be to make a deep copy of `other` before starting the iteration over items.
-  std::optional<std::string_view> delayed;
+  std::optional<DataProxy> delayed;
   // Note the inefficiency here: We are comparing some or all of the coords and
   // labels for each item. This could be improved by implementing the operations
   // for detail::DatasetData instead of DataProxy.
   for (const auto & [ name, item ] : a) {
```

Comment:
```suggestion
  for (const auto & [ [[maybe_unused]] name, item ] : a) {
```
and remove the `static_cast<void>(name);` below?

Reply:
don't know why it's suggesting to remove a space at the start of the line...

Reply:
That does not compile.